frontend/src/metabase/visualizations/components/settings/ChartSettingInputNumeric.tsx

```diff
@@ -14,20 +16,25 @@ const ChartSettingInputNumeric = ({
   value,
   ...props
 }: ChartSettingInputProps) => {
-  const [internalValue, setInternalValue] = useState(value);
+  const [internalValue, setInternalValue] = useState(value?.toString() ?? "");
 
   return (
     <ChartSettingNumericInput
-      type="number"
+      type="text"
       {..._.omit(props, "onChangeSettings")}
-      error={!!internalValue && isNaN(internalValue)}
+      error={internalValue !== "" && isNaN(Number(internalValue))}
       value={internalValue}
       onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
-        const num = parseFloat(e.target.value);
-        setInternalValue(num);
+        const everyCharValid = e.target.value
+          .split("")
+          .every(char => ALLOWED_CHARS.includes(char));
+
+        if (everyCharValid) {
+          setInternalValue(e.target.value);
+        }
       }}
       onBlur={(e: React.ChangeEvent<HTMLInputElement>) => {
-        const num = parseFloat(e.target.value);
+        const num = e.target.value !== "" ? Number(e.target.value) : Number.NaN;
```

Comment:
`parseFloat` allows invalid inputs with multiple decimals, wheras `Number` returns `NaN` for them.

<img width="223" alt="Screenshot 2023-08-09 at 10 40 20 AM" src="https://github.com/metabase/metabase/assets/37751258/7e922973-5e60-4540-9461-20822c23cfd0">